File: day6/day6.cpp

#include "../stdc++.h"
#include "../stringutils.h"

using namespace std;

void solution(vector<string>& input) {
    auto time = splitT<int>(split(input[0], ':')[1]);
    auto distance = splitT<int>(split(input[1], ':')[1]);

    for (auto t: time) {
        cout << t << endl;
    }

    int total = 1;

    // total *= len([i * (t - i) for i in range(1, t) if i * (t - i) > d])
    for (int i=1; i<time.size(); ++i) {
        int t = time[i];
        int d = i * (t - i);
        if (d > distance[i]) {
            total *= d;
        }
    }
    cout << total << endl;


}

int main(int argc, char** argv) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    ifstream file(argv[1]);

    if (!file.is_open()) {
        cout << "File not found" << endl;
        return 0;
    }

    vector<string> lines;
    string line;
    while (getline(file, line)) {
        lines.push_back(line);
    }

    solution(lines);

    file.close();
    return 0;
}

File: day6/day6.py

def solution(data):
    time = map(int, data[0].split(":")[1].split())
    distance = map(int, data[1].split(":")[1].split())

    total = 1
    for t, d in zip(time, distance):
        # i represents how much time we hold down
        # for i in range(1, t):
        #     distance_travelled = i * (t - i)
        #     if distance_travelled > d:
        #         distance += [distance_travelled]

        total *= len([i * (t - i) for i in range(1, t) if i * (t - i) > d])

    return total


if __name__ == "__main__":
    data = []
    with open("input2.txt", "r") as file:
        data = file.readlines()

    print(solution(data))